### PR TITLE
Allow injection of Pod repository annotations

### PR DESCRIPTION
**Pull request by ggimes**
- Updated the deployment template
- Updated Version
- Generated a new Helm chart
- Fixed the access rule file used for the demo mode for Oathkeeper version > 0.18

## Related issue
Github issue: [#278](https://github.com/ory/oathkeeper/issues/278)

## Proposed changes

In order to be able to inject Pod resource annotation, I need to update the `Deployment` template. This is, in my case, needed in order to make the readiness/liveness probes work properly with istio enabled (istio.io/docs/ops/app-health-check/#use-annotations-on-pod).

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [x] I have read the [security policy](../security/policy)
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation within the code base (if appropriate)

## Further comments

I have generated a new version of Oathkeeper helm chart (`helm package --app-version 0.0.29 --version 0.0.29 ./oathkeeper`), moved the archive in `docs/helm/charts` and regenerated the `index.yaml` (`helm repo index .`). Please don't hesitate to let me know if I miss something or if not the way we do that (I have 10 days experience with Helm ;-))

### CODE REVIEW

File: helm/charts/oathkeeper/templates/deployment.yaml [hunk 1]
```diff
@@ -8,6 +8,10 @@ metadata:
   {{- end }}
   labels:
 {{ include "oathkeeper.labels" . | indent 4 }}
+  {{- with .Values.deployment.annotations }}
```
[the remark this answers is not in the document]
Comment (ggimes) [reply helm/charts/oathkeeper/templates/deployment.yaml]: Sorry I can't see the link

[the remark this answers is not in the document]
Comment (aeneasr) [reply helm/charts/oathkeeper/templates/deployment.yaml]: It's this file line 23(before patch)/27(after patch) :) highlighted in yellow

image

File: helm/charts/oathkeeper/templates/deployment.yaml [hunk 2]
```diff
@@ -8,6 +8,7 @@ metadata:
...
```
Comment (aeneasr) [inline helm/charts/oathkeeper/templates/deployment.yaml]: This line can also be removed, right?

```suggestion
```

**Comment by aeneasr** [comment]
Thank you!

**Pull request merged**